OP_REVERSEBYTES implementation

sickpig opened the pull request:
Implementation of OP_REVERSEBYTES is very straightforward.

Previously named OP_REVERSE, then OP_BSWAP, then OP_ENDIAN_REVERSE.

Test Plan: Tests listed in the OP_REVERSEBYTES PR: EyeOfPython/bitcoincash.org:spec/op_reversebytes.md@master

This is a port of  reviews.bitcoinabc.org/D4871 and reviews.bitcoinabc.org/D5283

built on top of #2104

EyeOfPython committed:
Added OP_REVERSEBYTES+implementation, added (always disabled) activation flag

Implementation of OP_REVERSEBYTES is very straightforward.

Previously named OP_REVERSE, then OP_BSWAP, then OP_ENDIAN_REVERSE.

Test Plan: Tests listed in the OP_REVERSEBYTES PR: https://github.com/EyeOfPython/bitcoincash.org/blob/master/spec/op_reversebytes.md

This is a port of  https://reviews.bitcoinabc.org/D4871

sickpig commented:
for some weird reason travis get stuck in a reproducible way on the 2nd job:

```
Running tests: wallet_tests from ../../src/wallet/test/wallet_tests.cpp
Running tests: walletdb_tests from ../../src/wallet/test/walletdb_tests.cpp
Running tests: wallet_crypto from ../../src/wallet/test/crypto_tests.cpp
No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.
Check the details on how to adjust your build configuration on: https://docs.travis-ci.com/user/common-build-problems/#build-times-out-because-no-output-was-received
The build has been terminated
```

working on it

ptschip commented on qa/rpc-tests/test_framework/cashlib/cashlib.py:
```diff
@@ -286,6 +286,7 @@ class ScriptFlags(IntFlag):
     SCRIPT_DISALLOW_SEGWIT_RECOVERY = (1 << 20)
     SCRIPT_ENABLE_MUL_SHIFT_INVERT = (1 << 21)
+    SCRIPT_ENABLE_OP_REVERSEBYTES = (1 << 23)
 
```
why 1<<23 and not 1<<22  for SCRIPT_ENABLE_OP_REVERSEBYTES?

sickpig replied:
1U << 22 is reserved for SCRIPT_VERIFY_INPUT_SIGCHECKS (the other feature that will be added in the next update)

ptschip commented:
> for some weird reason travis get stuck in a reproducible way on the 2nd job:
> 
> ```
> Running tests: wallet_tests from ../../src/wallet/test/wallet_tests.cpp
> Running tests: walletdb_tests from ../../src/wallet/test/walletdb_tests.cpp
> Running tests: wallet_crypto from ../../src/wallet/test/crypto_tests.cpp
> No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.
> Check the details on how to adjust your build configuration on: https://docs.travis-ci.com/user/common-build-problems/#build-times-out-because-no-output-was-received
> The build has been terminated
> ```
> 
> working on it, see https://travis-ci.org/BitcoinUnlimited/BitcoinUnlimited/jobs/656986580#L2401

I would try it a few more times...I seems to remember those wallet tests sometimes getting hung up in the past when travis was having issues...they are quite intensive tests as I recall.

sickpig commented:
> > for some weird reason travis get stuck in a reproducible way on the 2nd job:
> > ```
> > Running tests: wallet_tests from ../../src/wallet/test/wallet_tests.cpp
> > Running tests: walletdb_tests from ../../src/wallet/test/walletdb_tests.cpp
> > Running tests: wallet_crypto from ../../src/wallet/test/crypto_tests.cpp
> > No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.
> > Check the details on how to adjust your build configuration on: https://docs.travis-ci.com/user/common-build-problems/#build-times-out-because-no-output-was-received
> > The build has been terminated
> > ```
> > 
> > 
> > working on it, see [travis-ci.org/BitcoinUnlimited/BitcoinUnlimited/jobs/656986580#L2401](https://travis-ci.org/BitcoinUnlimited/BitcoinUnlimited/jobs/656986580#L2401)
> 
> I would try it a few more times...I seems to remember those wallet tests sometimes getting hung up in the past when travis was having issues...they are quite intensive tests as I recall.

will give it another try

sickpig committed:
Move pad_tx from txtools.py to blocktools.py

Since that function was the whole content of txtools.py
we removed the file completely.

sickpig committed:
Add create_tx_with_script to blocktools.py

Return one-input, one-output transaction object spending the prevtx's n-th output with the given amount.
Can optionally pass scriptPubKey and scriptSig, default is anyone-can-spend output.

EyeOfPython committed:
OP_REVERSEBYTES activation logic

This is a port of https://reviews.bitcoinabc.org/D5283

sickpig commented:
ported also activation code and activation test. still need to fix the latter thou. still WIP

sickpig commented:
Taking out of wip, ready for review

sickpig commented:
dunno why but for some reason the arm32 job is not even able to finish the build step for `bitcoind`, it must be a temporary Travis issue. 

https://travis-ci.org/BitcoinUnlimited/BitcoinUnlimited/jobs/658779737

I will investigate further and try to reproduce locally in case the next restart will result in another timeout.

sickpig commented:
I tried to reproduce locally but I have no problem at all to cross build arm32 in a decent time even in a entry level machine
